morph.go

```diff
@@ -143,3 +176,95 @@ func computePendingMigrations(appliedMigrations []*models.Migration, sourceMigra
 
 	return pendingMigrations, nil
 }
+
+func breakDownMigrations(migrations []*models.Migration, direction models.Direction) ([]*models.Migration, map[string]*models.Migration, error) {
+	rollbackMigrations := make(map[string]*models.Migration)
+	toBeAppliedMigrations := make([]*models.Migration, 0)
+	for _, migration := range migrations {
+		if migration.Direction != direction {
+			rollbackMigrations[migration.Name] = migration
+			continue
+		}
+		toBeAppliedMigrations = append(toBeAppliedMigrations, migration)
+	}
+
+	for _, migration := range toBeAppliedMigrations {
+		_, ok := rollbackMigrations[migration.Name]
+		if !ok {
+			return nil, nil, fmt.Errorf("the rollback migration file for %s is missing", migration.RawName)
+		}
+	}
+
+	return toBeAppliedMigrations, rollbackMigrations, nil
+}
+
+func (m *Morph) ApplyDown(limit int) (int, error) {
+	appliedMigrations, err := m.driver.AppliedMigrations()
+	if err != nil {
+		return -1, err
+	}
+
+	sortedMigrations := reverseSortMigrations(appliedMigrations)
+	downMigrations, err := findDownScrips(sortedMigrations, m.source.Migrations())
+
+	steps := limit
+	if limit < 0 || len(sortedMigrations) < steps {
+		steps = len(sortedMigrations)
+	}
+
+	var applied int
+	for i := 0; i < steps; i++ {
+		start := time.Now()
+		migrationName := sortedMigrations[i].Name
+		m.config.Logger.Println(InfoLoggerLight.Sprint(formatProgress(fmt.Sprintf(migrationProgressStart, migrationName))))
+
+		down := downMigrations[migrationName]
+		if err := m.driver.Apply(down, true); err != nil {
+			return applied, err
```

Comment:
This block should behave more or less the same of the `up` error block. If we decide to rollback to the same on both, or just return on both